portal/portal-impl/impl/src/java/org/sakaiproject/portal/charon/handlers/SiteHandler.java

```diff
@@ -505,8 +506,14 @@ public void doSite(HttpServletRequest req, HttpServletResponse res, Session sess
 		PortalRenderContext rcontext = portal.startPageContext(siteType, title, site
 				.getSkin(), req, site);
 
-		if (toolId != null) {
-			rcontext.put("currentExpanded", true);
+		try {
+			PreferencesEdit prefs = PreferencesService.edit(session.getUserId());
+			ResourcePropertiesEdit props = prefs.getPropertiesEdit(org.sakaiproject.user.api.PreferencesService.SITENAV_PREFS_KEY);
+			props.addProperty(PortalConstants.PROP_CURRENT_EXPANDED, "true");
+			props.addProperty(PortalConstants.PROP_EXPANDED_SITE, siteId);
+			PreferencesService.commit(prefs);
+		} catch (Exception any) {
+			log.warn("Exception caught whilst setting {} property: {}", SELECTED_PAGE_PROP, any.toString());
```

Comment:
I'm trying to debug another JIRA and ending up over here. what's the reason for doing this on backend instead of on frontend (e.g., local storage)?